pkgs/development/python-modules/py-cid/default.nix

```diff
@@ -0,0 +1,56 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+, pythonOlder
+, pytest-runner
+, pytestCheckHook
+, base58
+, py-multibase
+, py-multicodec
+, morphys
+, py-multihash
+, hypothesis
+}:
+
+buildPythonPackage rec {
+  pname = "py-cid";
+  version = "0.3.0";
+  disabled = pythonOlder "3.5";
+
+  src = fetchFromGitHub {
+    owner = "ipld";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "sha256-aN7ee25ghKKa90+FoMDCdGauToePc5AzDLV3tONvh4U=";
+  };
+
+  patches = [
+    ./requirements.patch
```

Comment:
Please replace the patch with substituteInPlace.

Reply:
Sure.
Why is substituteInPlace preferable to a patch, especially when multiple lines need to be changed? A patch has the advantage of not silently doing nothing when the source changes too much.

Reply:
substituteInPlace is easier to update and does not add a file to the repo.

Reply:
> I grep the log for changed substituteInPlace.

Can you elaborate please? Are you using git grep? And what for?
Also how do you detect silent breakages similar to what is described in 1842c4aaa409e4731d355aa1f77ed817dd3939a2?

Reply:
I am basically greping the output if ``nix log /nix/store/...py-cid`` for warnings that a substituteInPlace was not found. Also for python does failures are usually not silent and old substituteInPlace just pile up.